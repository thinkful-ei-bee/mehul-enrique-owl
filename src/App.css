
.App{
    display:flex;
    flex-direction:row;
    
}

.ppl-List{
display:flex;
flex-direction:column;
border:5px solid black;
max-width:200px;
}

.chat-list{
display:flex;
flex-direction:column;
border:5px solid black;
max-width:200px;
}

.stage-list{
    display:flex;
    flex-direction:row;
    max-width:500px;
    border:5px solid black;

    
}
