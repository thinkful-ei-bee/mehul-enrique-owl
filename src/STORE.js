import participants from './participants.js';
import chatEvents from './chatEvents.js';


const STORE = {
    ppl_list: participants,
    chat_info:chatEvents 
    };
  

export default STORE;