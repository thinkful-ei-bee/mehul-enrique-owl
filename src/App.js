import React from 'react';
import Person from './Person.js'
import Chat from './Chat.js'
import Stage from './stage.js'

import './App.css';


function App(props) {


let chat_list = props.chatEvents;

const onStage_ppl=props.participants.filter(participant=>participant.onStage) 
const inSession=props.participants.filter(participant=>participant.inSession && !participant.onStage) 
const left=props.participants.filter(participant=>!participant.onStage&&!participant.inSession) 
let ppl_list = onStage_ppl.concat(inSession).concat(left)


// chat list
let mappedChatList = chat_list.map(obj => {
  

    for(let x = 0; x< ppl_list.length; x++)
    {   
        if(ppl_list[x].id === obj.participantId)
        {
            obj.avatar = ppl_list[x].avatar;
            obj.name = ppl_list[x].name;
            
            break;
        }
    }

    return {
        id:obj.id,
        type:obj.type,
        message:obj.message,
        time:obj.time,
        timestamp:obj.timestamp,
        avatar:obj.avatar,
        name:obj.name
    }
    } 
); 





return (
<main className='App'>
<ul className="ppl-List"> 
{ppl_list.map(obj => (
    <Person
    id={obj.id}
    name={obj.name}
    avatar={obj.avatar}
    inSession = {obj.inSession}
    onStage = {obj.onStage}
  />
          ))}
</ul>
<ul className="chat-list"> 
{mappedChatList.map(obj => (
    
    <Chat
    id={obj.id}
    type={obj.type}
    message={obj.message}
    time = {obj.time}
    timestamp = {obj.timestamp}
    avatar = {obj.avatar}
    name = {obj.name}
  />
          ))}
</ul>

<ul className="stage-list"> 
{onStage_ppl.map(obj => (
    
    <Stage
    avatar = {obj.avatar}
    name = {obj.name}
  />
          ))}
</ul>



</main>
 );
}

  	
export default App;