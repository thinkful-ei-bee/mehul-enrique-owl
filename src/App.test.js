import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import STORE from './STORE.js';

it('renders without crasing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <App participants={STORE.ppl_list} chatEvents={STORE.chat_info}  />,
    div,
  );
  ReactDOM.unmountComponentAtNode(div);
 
});


it('snapshot test a app  as expected', () => {
  const tree = renderer
    .create( <App participants={STORE.ppl_list} chatEvents={STORE.chat_info}  />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});