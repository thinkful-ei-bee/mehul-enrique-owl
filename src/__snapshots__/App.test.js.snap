// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test a chat message as expected 1`] = `
<main
  className="App"
>
  <ul
    className="ppl-List"
  >
    <li
      className="Person"
    >
      <img
        alt="avatar-text"
        className="avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <span>
        Koren Templeton
      </span>
      <div
        className="green-dot"
      />
      <p
        className="location"
      >
        On Stage
      </p>
    </li>
    <li
      className="Person"
    >
      <img
        alt="avatar-text"
        className="avatar"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
      <span>
        Ashla Attwool
      </span>
      <div
        className="green-dot"
      />
      <p
        className="location"
      >
        On Stage
      </p>
    </li>
    <li
      className="Person"
    >
      <img
        alt="avatar-text"
        className="avatar"
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
      <span>
        Caty Flucker
      </span>
      <div
        className="green-dot"
      />
      <p
        className="location"
      >
        In Session
      </p>
    </li>
    <li
      className="Person"
    >
      <img
        alt="avatar-text"
        className="avatar"
        src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
      />
      <span>
        Frank Runciman
      </span>
      <div
        className="green-dot"
      />
      <p
        className="location"
      >
        In Session
      </p>
    </li>
    <li
      className="Person"
    >
      <img
        alt="avatar-text"
        className="avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <span>
        Axe Kubicka
      </span>
      <div
        className="grey-dot"
      />
      <p
        className="location"
      >
        Left Session
      </p>
    </li>
  </ul>
  <ul
    className="chat-list"
  >
    <li
      className="Message"
    >
      <img
        alt="avatar-text"
        className="avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <h3>
        Axe Kubicka
      </h3>
      <p>
        7
        :
        50
      </p>
      <p>
        Hello world
      </p>
    </li>
    <li
      className="action"
    >
      <p>
        Caty Flucker
         gave a thumbs up 
      </p>
    </li>
    <li
      className="action"
    >
      <p>
        Frank Runciman
         gave a thumbs down 
      </p>
    </li>
    <li
      className="action"
    >
      <p>
        Koren Templeton
         raised their hand 
      </p>
    </li>
    <li
      className="action"
    >
      <p>
        Koren Templeton
         clapped 
      </p>
    </li>
    <li
      className="action"
    >
      <p>
        Koren Templeton
         joined 
      </p>
    </li>
    <li
      className="action"
    >
      <p>
        Ashla Attwool
         left 
      </p>
    </li>
    <li
      className="action"
    >
      <p>
        Axe Kubicka
         joined the stage 
      </p>
    </li>
    <li
      className="action"
    >
      <p>
        Axe Kubicka
         left the stage
      </p>
    </li>
  </ul>
  <ul
    className="stage-list"
  >
    <li
      className="stage-obj"
    >
      <p>
        Koren Templeton
         
      </p>
      <img
        alt="avatar-text"
        className="avatar-stage"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </li>
    <li
      className="stage-obj"
    >
      <p>
        Ashla Attwool
         
      </p>
      <img
        alt="avatar-text"
        className="avatar-stage"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
    </li>
  </ul>
</main>
`;
